tests/tizen_capi/unittest_tizen_capi_latency.cc

```diff
@@ -0,0 +1,280 @@
+  {
+    root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+    /* supposed to run test in build directory */
+    if (root_path == NULL)
+      root_path = "..";
+
+    status = ML_ERROR_NONE;
+    single_total_invoke_duration = 0;
+    single_invoke_duration_f = 0;
+    direct_invoke_duration_f = 0;
+
+    model_file = g_build_filename (
+        root_path, "tests", "test_models", "models", new_model, NULL);
+    ASSERT_TRUE (g_file_test (model_file, G_FILE_TEST_EXISTS));
+
+    data_file = g_build_filename (
+        root_path, "tests", "test_models", "data", "orange.raw", NULL);
+    ASSERT_TRUE (g_file_test (data_file, G_FILE_TEST_EXISTS));
+
+#if defined(ENABLE_NNFW_RUNTIME)
+    manifest_file = g_build_filename (root_path, "tests", "test_models",
+        "models", "metadata", "MANIFEST", NULL);
+    ASSERT_TRUE (g_file_test (manifest_file, G_FILE_TEST_EXISTS));
+
+    replace_command = g_strdup_printf ("sed -i '/%s/c\\\"models\" : [ \"%s\" ],' %s",
+        orig_model, new_model, manifest_file);
+    status = system (replace_command);
+    g_free (replace_command);
+    ASSERT_EQ (status, 0);
+#endif
+
+    fd = open (data_file, O_RDONLY);
+    EXPECT_TRUE (fd >= 0);
+  }
+
+  /**
+   * @brief final tear down overload virtual function
+   */
+  virtual void TearDown ()
+  {
+#if defined(ENABLE_NNFW_RUNTIME)
+    if (manifest_file) {
+      replace_command = g_strdup_printf ("sed -i '/%s/c\\\"models\" : [ \"%s\" ],' %s",
+          new_model, orig_model, manifest_file);
+      status = system (replace_command);
+      g_free (replace_command);
+    }
+    g_free (manifest_file);
+#endif
+
+    g_free (model_file);
+    g_free (data_file);
+
+    if (fd >= 0)
+      close (fd);
+  }
+
+  /**
+   * @brief get the framework internal latency
+   */
+  void extractInternalInvokeTime (const char *filter)
+  {
+    /** Find the framework */
+    const GstTensorFilterFramework *sp = nnstreamer_filter_find (filter);
+    EXPECT_NE (sp, (void *)NULL);
+
+    /** Extract the statictics from the framework */
+    EXPECT_TRUE (sp->statistics != NULL);
+    if (sp->statistics) {
+      direct_invoke_duration_f = (sp->statistics->total_invoke_latency * 1.0f) /
+        sp->statistics->total_invoke_num;
+
+      float internalOverheadTime = (sp->statistics->total_overhead_latency * 1.0f) /
+        sp->statistics->total_invoke_num;
+      g_warning ("Latency introduced by tensor filter extension = %f us.",
+          internalOverheadTime);
+    }
+  }
+
+  /**
+   * @brief get argmax from the array
+   */
+  void matchOutput (void *output_data, size_t size)
+  {
+    size_t idx, max_idx = 0;
+    guint8 *array = (guint8 *)output_data;
+    guint8 max_value = 0;
+    for (idx = 0; idx < size; idx++) {
+      if (max_value < array[idx]) {
+        max_idx = idx;
+        max_value = array[idx];
+      }
+    }
+
+    /**
+     * entry 952 (idx 951) is orange as per
+     * tests/test_models/labels/labels.txt
+     */
+    EXPECT_EQ (max_idx, 951);
```

Comment:
How about `951` -> `951U` ? I think compiling on Ubuntu would warn this.

Reply:
Updated.